### PR TITLE
[skip ci][home] add tracking transparency request to home

### PR DESCRIPTION
**Pull request by cruzach**
# Why

Since we include `NSUserTrackingUsageDescription` in Expo Go, we must request tracking permissions.

# How

# Test Plan

Only visible on iOS, and after granted or denying permissions, the item is no longer shown

image

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: home/screens/UserSettingsScreen.tsx [hunk 1]
```diff
@@ -104,6 +109,41 @@ function MenuGestureItem() {
...
+      const { status } = await Tracking.getTrackingPermissionsAsync();
+      setShowTrackingItem(status === 'undetermined');
+    })();
+  }, [showTrackingItem]);
+
+  return showTrackingItem ? (
+    <View style={styles.marginTop}>
+      <SectionHeader title="Tracking" />
+      <ListItem
+        title="Allow projects to track device data"
```
Comment (brentvatne) [inline home/screens/UserSettingsScreen.tsx:125]: add `last` prop here like on `SignOutItem` so the row has the correct bottom border

**Review by brentvatne — CHANGES_REQUESTED** [review]

File: home/screens/UserSettingsScreen.tsx (hunk 1 above)
Comment (cruzach) [inline home/screens/UserSettingsScreen.tsx:125]:
```suggestion
        last
        title="Allow projects to track device data"
```

File: home/screens/UserSettingsScreen.tsx [hunk 2]
```diff
@@ -104,6 +109,42 @@ function MenuGestureItem() {
...
+      setShowTrackingItem(status === 'undetermined');
+    })();
+  }, [showTrackingItem]);
+
+  return showTrackingItem ? (
+    <View style={styles.marginTop}>
+      <SectionHeader title="Tracking" />
+      <ListItem
+        last
+        title="Allow projects to track device data"
```
Comment (cruzach) [inline home/screens/UserSettingsScreen.tsx:126]:
```suggestion
        title="Allow projects to track app-related data"
```

File: home/screens/UserSettingsScreen.tsx [hunk 3]
```diff
@@ -104,6 +109,42 @@ function MenuGestureItem() {
...
+      setShowTrackingItem(status === 'undetermined');
+    })();
+  }, [showTrackingItem]);
+
+  return showTrackingItem ? (
+    <View style={styles.marginTop}>
+      <SectionHeader title="Tracking" />
+      <ListItem
+        last
+        title="Allow projects to track app-related data"
```
Comment (cruzach) [inline home/screens/UserSettingsScreen.tsx:126]:
```suggestion
        title="Allow access to app-related data for tracking"
```

File: packages/expo-tracking-transparency/src/TrackingTransparency.ts [hunk 4]
```diff
@@ -77,7 +77,11 @@ export async function getTrackingPermissionsAsync(): Promise<PermissionResponse>
  * resolve to `granted`.
  */
 export function isAvailable(): boolean {
-  return Platform.OS === 'ios' && parseInt(Platform.Version.toString(), 10) >= 14;
+  return (
+    Platform.OS === 'ios' &&
+    parseInt(Platform.Version.toString(), 10) >= 14 &&
+    ExpoTrackingTransparency
```
Comment (cruzach) [inline packages/expo-tracking-transparency/src/TrackingTransparency.ts:83]: `ExpoTrackingTransparency ` returns `undefined` if not available, or populated object if it is

**Review by brentvatne — APPROVED** [review]

**Pull request merged**